### ISSUE TITLE
Geth won't sync historical blocks

### ISSUE THREAD

**Issue by paymog** [ethereum/go-ethereum#26011]
#### System information

Geth version: 1.11.0-unstable
CL client & version: not running one
OS & Version: OSX
Commit hash : `15b7e0b25459786f4914590cc980039a7ea564f1`

#### Expected behaviour

I expect geth to sync blocks when it has enough peers

#### Actual behaviour

Geth connects to 3+ peers but never seems to get served any blocks. Here's the startup output:

```
 ❯❯❯ ./build/bin/geth --verbosity=4
INFO [10-19|12:09:40.309] Starting Geth on Ethereum mainnet...
INFO [10-19|12:09:40.310] Bumping default cache on mainnet         provided=1024 updated=4096
INFO [10-19|12:09:40.311] Maximum peer count                       ETH=50 LES=0 total=50
DEBUG[10-19|12:09:40.313] FS scan times                            list="46.25µs" set=250ns diff=667ns
DEBUG[10-19|12:09:40.313] FS scan times                            list="89.125µs" set=250ns diff=375ns
DEBUG[10-19|12:09:40.314] Sanitizing Go's GC trigger               percent=25
INFO [10-19|12:09:40.317] Set global gas cap                       cap=50,000,000
INFO [10-19|12:09:40.318] Allocated trie memory caches             clean=614.00MiB dirty=1024.00MiB
INFO [10-19|12:09:40.318] Allocated cache and file handles         database=/Users/paymahn/Library/Ethereum/geth/chaindata cache=2.00GiB handles=5120
DEBUG[10-19|12:09:40.449] Chain freezer table opened               database=/Users/paymahn/Library/Ethereum/geth/chaindata/ancient/chain table=headers items=0 size=0.00B
DEBUG[10-19|12:09:40.450] Chain freezer table opened               database=/Users/paymahn/Library/Ethereum/geth/chaindata/ancient/chain table=hashes  items=0 size=0.00B
DEBUG[10-19|12:09:40.451] Chain freezer table opened               database=/Users/paymahn/Library/Ethereum/geth/chaindata/ancient/chain table=bodies  items=0 size=0.00B
DEBUG[10-19|12:09:40.451] Chain freezer table opened               database=/Users/paymahn/Library/Ethereum/geth/chaindata/ancient/chain table=receipts items=0 size=0.00B
DEBUG[10-19|12:09:40.451] Chain freezer table opened               database=/Users/paymahn/Library/Ethereum/geth/chaindata/ancient/chain table=diffs    items=0 size=0.00B
INFO [10-19|12:09:40.451] Opened ancient database                  database=/Users/paymahn/Library/Ethereum/geth/chaindata/ancient/chain readonly=false
DEBUG[10-19|12:09:40.454] Current full block not old enough        number=0 hash=d4e567..cb8fa3 delay=90000
INFO [10-19|12:09:40.454] Disk storage enabled for ethash caches   dir=/Users/paymahn/Library/Ethereum/geth/ethash count=3
INFO [10-19|12:09:40.455] Disk storage enabled for ethash DAGs     dir=/Users/paymahn/Library/Ethash count=2
INFO [10-19|12:09:40.455] Initialising Ethereum protocol           network=1 dbversion=8
INFO [10-19|12:09:40.456]
INFO [10-19|12:09:40.456] ---------------------------------------------------------------------------------------------------------------------------------------------------------
INFO [10-19|12:09:40.456] Chain ID:  1 (mainnet)
INFO [10-19|12:09:40.456] Consensus: Beacon (proof-of-stake), merged from Ethash (proof-of-work)
INFO [10-19|12:09:40.456]
INFO [10-19|12:09:40.456] Pre-Merge hard forks:
INFO [10-19|12:09:40.456]  - Homestead:                   1150000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/homestead.md)
INFO [10-19|12:09:40.457]  - DAO Fork:                    1920000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/dao-fork.md)
INFO [10-19|12:09:40.457]  - Tangerine Whistle (EIP 150): 2463000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/tangerine-whistle.md)
INFO [10-19|12:09:40.457]  - Spurious Dragon/1 (EIP 155): 2675000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/spurious-dragon.md)
INFO [10-19|12:09:40.457]  - Spurious Dragon/2 (EIP 158): 2675000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/spurious-dragon.md)
INFO [10-19|12:09:40.457]  - Byzantium:                   4370000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/byzantium.md)
INFO [10-19|12:09:40.457]  - Constantinople:              7280000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/constantinople.md)
INFO [10-19|12:09:40.457]  - Petersburg:                  7280000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/petersburg.md)
INFO [10-19|12:09:40.457]  - Istanbul:                    9069000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/istanbul.md)
INFO [10-19|12:09:40.457]  - Muir Glacier:                9200000  (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/muir-glacier.md)
INFO [10-19|12:09:40.457]  - Berlin:                      12244000 (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/berlin.md)
INFO [10-19|12:09:40.457]  - London:                      12965000 (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/london.md)
INFO [10-19|12:09:40.457]  - Arrow Glacier:               13773000 (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/arrow-glacier.md)
INFO [10-19|12:09:40.457]  - Gray Glacier:                15050000 (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/gray-glacier.md)
INFO [10-19|12:09:40.457]
INFO [10-19|12:09:40.457] Merge configured:
INFO [10-19|12:09:40.457]  - Hard-fork specification:    https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/paris.md
INFO [10-19|12:09:40.457]  - Network known to be merged: true
INFO [10-19|12:09:40.457]  - Total terminal difficulty:  58750000000000000000000
INFO [10-19|12:09:40.457]  - Merge netsplit block:       <nil>
INFO [10-19|12:09:40.457] ---------------------------------------------------------------------------------------------------------------------------------------------------------
INFO [10-19|12:09:40.457]
INFO [10-19|12:09:40.469] Loaded most recent local header          number=0 hash=d4e567..cb8fa3 td=17,179,869,184 age=53y6mo3w
INFO [10-19|12:09:40.469] Loaded most recent local full block      number=0 hash=d4e567..cb8fa3 td=17,179,869,184 age=53y6mo3w
INFO [10-19|12:09:40.469] Loaded most recent local fast block      number=0 hash=d4e567..cb8fa3 td=17,179,869,184 age=53y6mo3w
DEBUG[10-19|12:09:40.470] Reinjecting stale transactions           count=0
INFO [10-19|12:09:40.470] Loaded local transaction journal         transactions=0 dropped=0
INFO [10-19|12:09:40.471] Regenerated local transaction journal    transactions=0 accounts=0
INFO [10-19|12:09:40.472] Chain post-merge, sync via beacon client
INFO [10-19|12:09:40.472] Gasprice oracle is ignoring threshold set threshold=2
WARN [10-19|12:09:40.472] Engine API enabled                       protocol=eth
INFO [10-19|12:09:40.473] Starting peer-to-peer node               instance=Geth/v1.11.0-unstable-15b7e0b2-20221018/darwin-arm64/go1.19.2
DEBUG[10-19|12:09:40.591] UDP listener up                          addr=[::]:30303
DEBUG[10-19|12:09:40.598] TCP listener up                          addr=[::]:30303
INFO [10-19|12:09:40.601] New local node record                    seq=1,666,122,124,860 id=3b9b1e3a7141f930 ip=127.0.0.1 udp=30303 tcp=30303
INFO [10-19|12:09:40.602] Started P2P networking                   self=enode://3a58de99d66c5155d7ec74b4aab6f0f1da26f03428b2a154072d8caf10650e26bfe913f05e40db5852eea7e3a3073778755fcef2c4ec4461362d08c34f6c514a@127.0.0.1:30303
DEBUG[10-19|12:09:40.603] IPCs registered                          namespaces=admin,debug,web3,eth,txpool,personal,ethash,miner,net,engine
INFO [10-19|12:09:40.604] IPC endpoint opened                      url=/Users/paymahn/Library/Ethereum/geth.ipc
INFO [10-19|12:09:40.606] Loaded JWT secret file                   path=/Users/paymahn/Library/Ethereum/geth/jwtsecret crc32=0x7d52ee5
DEBUG[10-19|12:09:40.608] Allowed origin(s) for WS RPC interface [localhost]
INFO [10-19|12:09:40.658] WebSocket enabled                        url=ws://127.0.0.1:8551
INFO [10-19|12:09:40.658] HTTP server started                      endpoint=127.0.0.1:8551 auth=true prefix= cors=localhost vhosts=localhost
INFO [10-19|12:09:43.119] New local node record                    seq=1,666,122,124,861 id=3b9b1e3a7141f930 ip=190.246.124.15 udp=30303 tcp=30303
INFO [10-19|12:09:43.470] Mapped network port                      proto=tcp extport=30303 intport=30303 interface="UPNP IGDv1-IP1"
INFO [10-19|12:09:43.974] Mapped network port                      proto=udp extport=30303 intport=30303 interface="UPNP IGDv1-IP1"
```

Here's geth connecting to more than 3 peers:

```
DEBUG[10-19|12:18:07.218] Revalidated node                         b=7  id=3be9d086c06f94cd checks=1
DEBUG[10-19|12:18:08.347] Adding p2p peer                          peercount=6 id=39bc67d31244a004 conn=inbound addr=202.28.119.38:39498   name=Geth/v1.10.1-unstabl...
DEBUG[10-19|12:18:08.350] Ethereum handshake failed                id=39bc67d31244a004 conn=inbound err="network ID mismatch: 17 (!= 1)"
DEBUG[10-19|12:18:08.352] Message handling failed in `snap`        peer=39bc67d3 err=EOF
DEBUG[10-19|12:18:08.355] Removing p2p peer                        peercount=5 id=39bc67d31244a004 duration=7.938ms     req=false err="network ID mismatch: 17 (!= 1)"
DEBUG[10-19|12:18:08.897] Adding p2p peer                          peercount=6 id=4ba7d57e3429a5db conn=dyndial addr=167.235.12.240:30303  name=erigon/v2.28.0-stabl...

DEBUG[10-19|12:18:09.125] Ethereum handshake failed                id=4ba7d57e3429a5db conn=dyndial err=EOF
DEBUG[10-19|12:18:09.125] Removing p2p peer                        peercount=5 id=4ba7d57e3429a5db duration=228.329ms   req=true  err="disconnect requested"
DEBUG[10-19|12:18:09.195] Adding p2p peer                          peercount=6 id=2202186b3755df1d conn=dyndial addr=167.86.91.151:30303   name=Geth/v1.0.0-alpha1-3...
DEBUG[10-19|12:18:09.427] Ethereum handshake failed                id=2202186b3755df1d conn=dyndial err="network ID mismatch: 9372 (!= 1)"
DEBUG[10-19|12:18:09.427] Removing p2p peer                        peercount=5 id=2202186b3755df1d duration=231.580ms   req=false err="network ID mismatch: 9372 (!= 1)"
DEBUG[10-19|12:18:09.868] Adding p2p peer                          peercount=6 id=ad5baa0d4afeb22a conn=inbound addr=34.64.210.151:45778   name=Geth/v1.10.23-stable...
DEBUG[10-19|12:18:09.868] Ethereum handshake failed                id=ad5baa0d4afeb22a conn=inbound err="network ID mismatch: 10001 (!= 1)"
DEBUG[10-19|12:18:09.868] Removing p2p peer                        peercount=5 id=ad5baa0d4afeb22a duration="293.084µs" req=false err="network ID mismatch: 10001 (!= 1)"
```

However, geth never gets any blocks served to it. I notice that sometimes the following log line is printed:

```
Current full block not old enough        number=0 hash=d4e567..cb8fa3 delay=90000
```

#### Steps to reproduce the behaviour

I don't know how to repro this on other folks machines. I pulled checked out commit `15b7e0b25459786f4914590cc980039a7ea564f1`, built geth and then started.

**Comment by holiman** [comment]
It is not possible to sync Ethereum mainnet using only `geth` nowadays, post-merge. It requires that you also run a beacon-node, a.k.a CL client. Such as lighthouse or prysm or teku.

**Comment by paymog** [comment]
Got it! For anyone else who comes across this, it seems like your consensus client also needs to be fully synced before historical blocks will be downloaded. I tried running Prysm but even with Prysm running Geth wouldn't sync. Then I tried lighthouse with a checkpoint sync and now it seems that Geth is downloading beacon headers and I suspect after those finish downloading it will start syncing historical blocks as well. @holiman does that sound correct to you?
